pyro/ops/provenance.py

```diff
@@ -0,0 +1,71 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+
+
+class ProvenanceTensor(torch.Tensor):
+    """
+    Provenance tracking implementation in Pytorch.
```

Comment:
Maybe add slightly more documentation here explaining what extra metadata is attached and a reference to 'Wingate, David, et al. "Nonstandard Interpretations of Probabilistic Programs for Efficient Inference." NIPS. 2011'?

Reply:
done